sympy/integrals/rationaltools.py

```diff
@@ -219,7 +219,7 @@ def ratint_logpart(f, g, x, t=None):
         R_map[r.degree()] = r
 
     def _include_sign(c, sqf):
-        if (c < 0) == True:
+        if c.is_real and (c < 0) == True:
```

Comment:
This should exclude `c = -oo` but is that the only case where `c.is_real` would be needed?

Reply:
This seems to be necessary as comparisons will raise an error for manifestly non-real arguments. Was there some discussion of possibly returning False?

Reply:
I did not consider the extended_real case. Forgot about that, but I think it makes sense, especially here. (The other cases seems to me more like they are rewrites for specific combinations, so the generic approach should work, although it may be better to change it there as well.)

I have not seen a discussion about returning False (or None) as far as I recall, more the general discussion about comparisons with complex values, with the observation that you state about the errors.

Reply:
#16956 changes this kind of comparison so that it only evaluates when both sides are known to be extended_real.

Reply:
OK! It seems like that is about to get finished quite soon, so I put this in hold and see to what extent this is required once that PR is merged.